src/http-client.js

```diff
@@ -0,0 +1,218 @@
+
+        this.axiosInstance.interceptors.request.use(gzipRequest, (e) => {
+            return Promise.reject(e);
+        });
+    }
+
+    async call(callOptions) {
+        const mergedOptions = { ...this.defaultOptions, ...callOptions };
+        const normalizedOptions = this._validateAndNormalizeOptions(mergedOptions);
+        return this._call(normalizedOptions);
+    }
+
+    _validateAndNormalizeOptions(options) {
+        /* eslint-disable prefer-const */
+        let {
+            baseUrl,
+            method,
+            token,
+            expBackoffMillis = EXP_BACKOFF_MILLIS,
+            expBackoffMaxRepeats = EXP_BACKOFF_MAX_REPEATS,
+            retryOnStatusCodes = [RATE_LIMIT_EXCEEDED_STATUS_CODE],
+        } = options;
+
+
+        if (typeof baseUrl !== 'string') {
+            throw new ApifyClientError(INVALID_PARAMETER_ERROR_TYPE, 'The "options.baseUrl" parameter of type string is required.');
+        }
+        // Remove trailing forward slash from baseUrl.
+        if (baseUrl.substr(-1) === '/') baseUrl = baseUrl.slice(0, -1);
+
+        if (typeof method !== 'string') {
+            throw new ApifyClientError(INVALID_PARAMETER_ERROR_TYPE, 'The "options.method" parameter of type string is required.');
+        }
+        method = method.toUpperCase();
+        if (!ALLOWED_HTTP_METHODS.includes(method)) {
+            throw new ApifyClientError(
+                INVALID_PARAMETER_ERROR_TYPE,
+                'The "options.method" parameter is invalid. '
+                + `Expected one of ${ALLOWED_HTTP_METHODS.join(', ')} but got: ${options.method}`,
+            );
+        }
+
+        if (token && typeof token !== 'string') {
+            throw new ApifyClientError(INVALID_PARAMETER_ERROR_TYPE, 'The "options.token" parameter must be a string.');
+        }
+
+        return {
+            ...options,
+            baseUrl,
+            method,
+            expBackoffMillis,
+            expBackoffMaxRepeats,
+            retryOnStatusCodes,
+        };
+    }
+
+    _getAxiosConfig(options) {
+        return {
+            baseURL: `${options.baseUrl}${options.basePath}`,
+            url: options.url,
+            method: options.method,
+            headers: {
+                'user-agent': CLIENT_USER_AGENT,
+                'accept-encoding': 'gzip',
+                'content-type': 'application/json; charset=utf-8',
+                ...options.headers,
+            },
+            params: {
+                ...options.qs,
+                token: options.token,
+            },
+            data: options.body,
+            // Override the default transform function to get unparsed response.
+            transformResponse: res => res,
+            httpAgent: this.httpAgent,
+            httpsAgent: this.httpsAgent,
+            validateStatus: null,
+            responseType: options.encoding === null ? 'arraybuffer' : null,
+        };
+    }
+
+    _getResponseLike(axiosResponse) {
```

Comment:
I'd add a comment here that one part handles server-side and the latter one browser response